pkg/controllers/provisioning/scheduling/suite_test.go

```diff
@@ -3386,6 +3387,222 @@ var _ = Context("NodePool", func() {
 			})
 		})
 	})
+
+	Describe("Deleting Nodes", func() {
+		It("should re-schedule pods from a deleting node when pods are active", func() {
+			ExpectApplied(ctx, env.Client, nodePool)
+			pod := test.UnschedulablePod(
+				test.PodOptions{ResourceRequirements: v1.ResourceRequirements{
+					Requests: map[v1.ResourceName]resource.Quantity{
+						v1.ResourceMemory: resource.MustParse("100M"),
+					},
+				}})
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
+			node := ExpectScheduled(ctx, env.Client, pod)
+			Expect(node.Labels[v1.LabelInstanceTypeStable]).To(Equal("small-instance-type"))
+
+			// Mark for deletion so that we consider all pods on this node for reschedulability
+			cluster.MarkForDeletion(node.Spec.ProviderID)
+
+			// Trigger a provisioning loop and expect another node to get created
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov)
+
+			nodes := ExpectNodes(ctx, env.Client)
+			Expect(nodes).To(HaveLen(2))
+
+			// Expect both nodes to be of the same size to schedule the pod once it gets re-created
+			for _, n := range nodes {
+				Expect(n.Labels[v1.LabelInstanceTypeStable]).To(Equal("small-instance-type"))
+			}
+		})
+		It("should not re-schedule pods from a deleting node when pods are not active", func() {
+			ExpectApplied(ctx, env.Client, nodePool)
+			pod := test.UnschedulablePod(
+				test.PodOptions{ResourceRequirements: v1.ResourceRequirements{
+					Requests: map[v1.ResourceName]resource.Quantity{
+						v1.ResourceMemory: resource.MustParse("100M"),
+					},
+				}})
+			ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, prov, pod)
+			node := ExpectScheduled(ctx, env.Client, pod)
+			Expect(node.Labels[v1.LabelInstanceTypeStable]).To(Equal("small-instance-type"))
+
+			// Mark for deletion so that we consider all pods on this node for reschedulability
+			cluster.MarkForDeletion(node.Spec.ProviderID)
+
+			// Trigger an eviction to set the deletion timestamp but not delete the pod
+			Expect(env.KubernetesInterface.PolicyV1().Evictions(pod.Namespace).Evict(ctx, &policyv1.Eviction{
+				ObjectMeta: metav1.ObjectMeta{Name: pod.Name, Namespace: pod.Namespace},
+			})).To(Succeed())
+			ExpectEvicted(ctx, env.Client, pod)
+			ExpectExists(ctx, env.Client, pod)
```

Comment:
Same here on putting the eviction call in the ExpectEvicted() block. It seems like ExpectEvicted is almost similar if not the same as a ExpectDeleting(obj)